hooks/boot/13-tuning.rb

```diff
@@ -4,10 +4,13 @@
   TUNING_SIZES = ['default'] + ['medium', 'large', 'extra-large', 'extra-extra-large']
   TUNING_FACT = 'tuning'.freeze
 
-  app_option(
-    '--tuning',
-    'INSTALLATION_SIZE',
-    "Tune for an installation size. Choices: #{TUNING_SIZES.join(', ')}",
-    :default => get_custom_fact(TUNING_FACT)
-  )
+  app_option('--tuning', 'INSTALLATION_SIZE', "Tune for an installation size. Choices: #{TUNING_SIZES.join(', ')}") do |value|
+    unless TUNING_SIZES.include?(value)
+      puts "Invalid option supplied for --tuning. Please choose from one of: #{TUNING_SIZES.join(', ')}"
+      puts "See: 'foreman-installer --help'"
```

Comment:
Can this be the binary name the user called? So ARGV[0], $0 or $PROGRAM_NAME or whatever Ruby likes today.

(Thinking branding ;))

Reply:
Good call, that led me to using a native method to handle this:

```
      raise Clamp::UsageError.new("Invalid option supplied for --tuning. Please choose from one of: #{TUNING_SIZES.join(', ')}", self)
```

Now this does directly reference Clamp which we try to avoid as far as I know. So, I can do one of two things or a combination of both:

 1) Introduce a Kafo::UsageError that inherits Clamp::UsageError
 2) Introduce a hook method `raise_usage_error` in Kafo that calls Clamp::UsageError

#2 feels like the best design while adhering to our creating a hook interface, but I can see an argument for the freedom to easily just reference native clamp exceptions. I'll hold https://github.com/theforeman/kafo/pull/334 until we decide since if I go the Kafo route that would be nice to include.

Reply:
Turns out that trying to call something in the Hook context won't exactly work because when the code is executed its executed within Clamps' environment. There is a method inside Clamps' environment named `signal_usage_error` that can be used. So I think the updated set of options are:

 1) Use Clamp::UsageError directly
 2) Use Clamp's `signal_usage_error`
 3) Define a Kafo::UsageError that inherits Clamp::UsageError and raise the Kafo version

Reply:
I think calling `signal_usage_error` is clean enough. It's an understandable API that doesn't mention clamp directly. I doubt they'll change it in a minor version and Kafo already depends on specific clamp versions so it's safe to expect it to remain unchanged. Perhaps we do need some tests in Kafo itself to ensure it works but that is more of a regression test and doesn't need a new release.

Reply:
Excellent, I have gone with signal_usage_error